docs/source/notes/extending.rst

```diff
@@ -14,59 +14,104 @@ Extending :mod:`torch.autograd`
 
 Adding operations to :mod:`~torch.autograd` requires implementing a new
 :class:`Function` subclass for each operation. Recall that :class:`Function` s
-are what :mod:`~torch.autograd` uses to compute the results and gradients, and
-encode the operation history. Every new function requires you to implement 2 methods:
-
-- :meth:`~Function.forward` - the code that performs the operation. It can take
+are what :mod:`~torch.autograd` uses to encode the operation history and compute
+gradients.
+
+When to use
+^^^^^^^^^^^
+In general, implement a custom function if you want to perform computations in your model
+that are not differentiable or rely on non-Pytorch libraries (e.g., NumPy), but
+still wish for your operation to chain with other ops and work with the autograd engine.
+
+In some situations, custom functions can also be used to improve performance and
+memory usage: If you implemented your forward and backward passes using a C++
+extension, you can wrap them in :class:`~Function` to interface with the autograd
+engine. If you'd like to reduce the number of buffers saved for the backward pass,
```

Comment:
Should we link the c++ extension tutorial showing how to do that here?

Reply:
linked!